File: test-swp.py
import urllib.request
from bs4 import BeautifulSoup

url = "https://immo.swp.de/suche/wohnungen-mieten/stuttgart-stadt"
req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
res = urllib.request.urlopen(req)
soup = BeautifulSoup(res.read(), 'html.parser')

items = soup.find_all('div', class_='featured-listings__item')
print(f"Found {len(items)} items")

for item in items[:2]:
    title_el = item.find('div', class_='featured-listings__item__title')
    title = title_el.text.strip() if title_el else "No title"
    print("Title:", title)
    
    price_el = item.find('p', class_='featured-listings__item__price')
    price = price_el.text.strip() if price_el else "No price"
    print("Price:", price)
    
    link_el = item.find('a')
    url = link_el['href'] if link_el else ''
    print("Link:", url)

File: test-swp.js
const cheerio = require('cheerio');

async function test() {
  const url = 'https://immo.swp.de/suche/wohnungen-mieten/stuttgart';
  console.log("Fetching", url);
  const res = await fetch(url, { headers: { 'User-Agent': 'Mozilla/5.0' }});
  const html = await res.text();
  const $ = cheerio.load(html);
  
  const properties = [];
  $('.list-item').each((_, el) => {
    const title = $(el).find('h2').text().trim();
    if(title) properties.push(title);
  });
  console.log("Found:", properties.length);
  if(properties.length === 0) {
     console.log("HTML Sample:", html.substring(0, 500));
  }
}
test();
